components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/soaptorest/template/RESTToSOAPMsgTemplate.java

```diff
@@ -0,0 +1,133 @@
+     *
+     * @param mapping soap to rest mapping json
+     * @param method http method for the resource
+     * @param soapAction soap action for the operation
+     * @param namespace soap namespace for the sequence
+     * @return in sequence string
+     */
+    public String getMappingInSequence( Map<String, String> mapping, String method, String soapAction, String namespace, JSONArray array) {
+
+        ConfigContext configcontext = new SOAPToRESTConfigContext(mapping, method, soapAction, namespace, array);
+        StringWriter writer = new StringWriter();
+        try {
+            VelocityContext context = configcontext.getContext();
+            context.internalGetKeys();
+
+            VelocityEngine velocityengine = new VelocityEngine();
+            if (!"not-defined".equalsIgnoreCase(getVelocityLogger())) {
+                velocityengine.setProperty( RuntimeConstants.RUNTIME_LOG_LOGSYSTEM_CLASS,
+                        "org.apache.velocity.runtime.log.Log4JLogChute" );
+                velocityengine.setProperty( "runtime.log.logsystem.log4j.logger", getVelocityLogger());
+            }
+
+            velocityengine.init();
+            Template t = velocityengine.getTemplate(this.getInSeqTemplatePath());
+
+            t.merge(context, writer);
+        } catch (Exception e) {
+            log.error("Velocity Error", e);
+        }
+        return writer.toString();
+    }
+
+    /**
+     * gets out sequence for the soap operation
+     *
+     * @return out sequence for the converted soap operation
+     */
+    public String getMappingOutSequence() {
+
+        ConfigContext configcontext = new SOAPToRESTConfigContext();
+        StringWriter writer = new StringWriter();
+        try {
+            VelocityContext context = configcontext.getContext();
+            context.internalGetKeys();
+
+            VelocityEngine velocityengine = new VelocityEngine();
+            if (!"not-defined".equalsIgnoreCase(getVelocityLogger())) {
+                velocityengine.setProperty( RuntimeConstants.RUNTIME_LOG_LOGSYSTEM_CLASS,
+                        "org.apache.velocity.runtime.log.Log4JLogChute" );
+                velocityengine.setProperty( "runtime.log.logsystem.log4j.logger", getVelocityLogger());
+            }
+
+            velocityengine.init();
+            Template t = velocityengine.getTemplate(this.getOutSeqTemplatePath());
+
+            t.merge(context, writer);
+        } catch (Exception e) {
+            log.error("Velocity Error", e);
+        }
+        return writer.toString();
+    }
+
+    private String getInSeqTemplatePath() {
+        return "repository" + File.separator + "resources" + File.separator + "api_templates" + File.separator + IN_SEQ_TEMPLATE_FILE + ".xml";
+    }
+
+    private String getOutSeqTemplatePath() {
+        return "repository" + File.separator + "resources" + File.separator + "api_templates" + File.separator + OUT_SEQ_TEMPLATE_FILE + ".xml";
```

Comment:
120 line limit seem to be violated

Reply:
fixed in e18f43f5dee9ed43a3239af62f38d67c72c78fa2